lib/services/sku-pack-service.js

```diff
@@ -477,63 +497,69 @@ function skuPackServiceFactory(
     SkuPackService.prototype.installPack = function(fromRoot, skuid) {
         var self = this;
         var contents;
-        var dst;
-        var installSKU;
         return fs.readFileAsync(fromRoot + '/config.json')
-            .then(function(fileContents) {
-                return [ fileContents, self.validatePack(fileContents, fromRoot) ];
-            })
-            .spread(function(fileContents) {
-                contents = fileContents;
-                if(skuid === undefined)  {
-                    var conf = JSON.parse(contents);
-                    if(!conf.hasOwnProperty('rules') || !conf.hasOwnProperty('name')) {
-                        throw new Error('rules or name is missing');
+        .then(function(fileContents) {
+            contents = fileContents;
+            return self.validatePack(fileContents, fromRoot, {
+                validateSku: !_.isUndefined(skuid)
+            });
+        })
+        .then(function() {
+            if(skuid === undefined)  {
```

Comment:
The controller code validates that the skuid exists if it is defined.  It is already validated.